Mysterious segmentation error upon running goa gen on API level error.

We declare several errors at the API level as follows, based on the documentation.

```
var _ = API("api", func() {

	// Generic errors that all services can throw
	Error("request_timed_out", func() {
		Description("The request took too long to complete and timed out.")
		Timeout()
	})
	Error("invalid_argument", func() {
		Description("The request contains an invalid argument.")
	})
	Error("not_found", func() {
		Description("The requested resource was not found.")
	})
...

	// GRPC error mappings
	GRPC(func() {
		Response("request_timed_out", CodeDeadlineExceeded)
		Response("invalid_argument", CodeInvalidArgument)
		Response("not_found", CodeNotFound)
		...
	})

	// HTTP error mappings
	HTTP(func() {
		Response("request_timed_out", StatusRequestTimeout)
		Response("invalid_argument", StatusBadRequest)
		Response("not_found", StatusNotFound)
		...
	})
```

And we refer to them with a service, as follows:
```
var _ = Service("service", func() {
	// TODO: Figure out why this is causing a null ptr error

	Error("request_timed_out")
	Error("invalid_argument")
	Error("not_found")
	...
```

Upon running goa gen, this happens:
```
goa gen portal/design
exit status 2
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x680746]
...
```

I've narrowed down the cause of the error to the service references themselves; commenting them causes the generation to work without issue. However, this is most likely due to none of the API errors being generated at all.

--- a/expr/grpc.go
+++ b/expr/grpc.go
@@ -37,3 +37,10 @@ func (g *GRPCExpr) ServiceFor(s *ServiceExpr) *GRPCServiceExpr {
 func (*GRPCExpr) EvalName() string {
 	return "API GRPC"
 }
+
+// Prepare initializes the error responses.
+func (g *GRPCExpr) Prepare() {
+	for _, er := range g.Errors {
+		er.Response.Prepare()
+	}
+}

--- a/expr/grpc_endpoint_test.go
+++ b/expr/grpc_endpoint_test.go
@@ -44,6 +44,10 @@ service "Service" gRPC endpoint "Method": field number 2 in attribute "key_dup_i
 			DSL:    testdata.GRPCEndpointWithExtendedTypes,
 			Errors: []string{},
 		},
+		"endpoint-with-inherit-error": {
+			DSL:    testdata.GRPCEndpointWithInheritErrorDSL,
+			Errors: []string{},
+		},
 	}
 	for name, c := range cases {
 		t.Run(name, func(t *testing.T) {

--- a/expr/testdata/endpoint_dsls.go
+++ b/expr/testdata/endpoint_dsls.go
@@ -713,3 +713,20 @@ var GRPCEndpointWithExtendedTypes = func() {
 		})
 	})
 }
+
+var GRPCEndpointWithInheritErrorDSL = func() {
+	API("API", func() {
+		Error("not_found")
+		GRPC(func() {
+			Response("not_found", CodeNotFound)
+		})
+	})
+	Service("Service", func() {
+		Error("not_found")
+		Method("Method", func() {
+			GRPC(func() {
+				Response(CodeOK)
+			})
+		})
+	})
+}

